core-web/libs/edit-content/src/lib/feature/edit-content/store/edit-content.store.ts

```diff
@@ -1,203 +1,307 @@
+                                        contentlet.inode,
+                                        DotRenderMode.EDITING
+                                    ),
+                                    contentlet: of(contentlet)
                                 });
-                            }
-                        )
-                    );
-                })
-            );
-        }
-    );
-
-    /**
-     * Fire the workflow action and update the contentlet and actions
-     *
-     * @private
-     * @param {(DotFireActionOptions<{ [key: string]: string | object }>)} options
-     * @return {*}  {Observable<{
-     *         actions: DotCMSWorkflowAction[];
-     *         contentlet: DotCMSContentlet;
-     *     }>}
-     * @memberof DotEditContentStore
-     */
-    private fireWorkflowAction(
-        options: DotFireActionOptions<{ [key: string]: string | object }>
-    ): Observable<{
-        actions: DotCMSWorkflowAction[];
-        contentlet: DotCMSContentlet;
-    }> {
-        return this.WorkflowActionsFireService.fireTo(options).pipe(
-            tap((contentlet) => {
-                if (!contentlet.inode) {
-                    this.router.navigate(['/c/content']);
+                            }),
+                            tapResponse({
+                                next: ({ contentlet, actions }) => {
+                                    router.navigate(['/content', contentlet.inode], {
+                                        replaceUrl: true,
+                                        queryParamsHandling: 'preserve'
+                                    });
+
+                                    patchState(store, {
+                                        contentlet,
+                                        actions,
+                                        status: ComponentStatus.LOADED,
+                                        error: null
+                                    });
+                                },
+                                error: (error: HttpErrorResponse) => {
+                                    patchState(store, {
+                                        status: ComponentStatus.ERROR,
+                                        error: 'Error firing workflow action'
+                                    });
+                                    dotHttpErrorManagerService.handle(error);
+                                }
+                            })
+                        );
+                    })
+                )
+            ),
+
+            /**
+             * Toggles the visibility of the sidebar by updating the application state
+             * and persists the sidebar's state to ensure consistency across sessions.
+             */
+            toggleSidebar: () => {
+                patchState(store, { showSidebar: !store.showSidebar() });
+                setPersistSidebarState(store.showSidebar() as unknown as string);
```

Comment:
@oidacra `setPersistSidebarState` take a string representing whether the sidebar is open or closed, instead to force typing should convert to a string with `store.showSidebar().toSting()` or allow setPersistSidebarState to take `string | boolean.`

Reply:
fixed